astropy/coordinates/matrix_utilities.py

```diff
@@ -62,9 +62,11 @@ def rotation_matrix(angle, axis='z', unit=None):
         A unitary rotation matrix.
     """
     if unit is None:
-        unit = u.degree
-
-    angle = Angle(angle, unit=unit)
+        angle = np.deg2rad(angle)
```

Comment:
Hmm, I appreciate the speed-up, but wonder about legibility. In particular, I now really have to think what happens when `angle` is already a Quantity. This branch will do the right thing, but it will be rather slow compared to just getting the Quantity in radians. And the second branch will fail in the admittedly unlikely case that someone, mistakenly, passes in a quantity not in radians and sets `unit=u.rad` (admittedly unlikely!).

That said, the current default action when one passes in a `Quantity` is quite awful: first convert to degrees then to radian...

How about this instead?
```
if isinstance(angle, u.Quantity):
    angle = angle.to_value(u.radian)
else:
    unit = u.Unit(unit) if unit is not None else u.degree
    angle = unit.to(u.rad, angle)
    # the above is 1 us if unit equals u.rad; does not
    # seem worth optimizing away.
```